src/Shared/LayoutRenderers/AspNetAppBasePathLayoutRenderer.cs

```diff
@@ -23,80 +22,97 @@ namespace NLog.Web.LayoutRenderers
     /// </summary>
 #else
     /// <summary>
-    /// Rendering Application BasePath. <see cref="HostingEnvironment.MapPath"/>("~")
+    /// Rendering Application BasePath. <see cref="IHostEnvironment.MapPath"/>("~")
     /// </summary>
 #endif
     [LayoutRenderer("aspnet-appbasepath")]
     [ThreadAgnostic]
     public class AspNetAppBasePathLayoutRenderer : LayoutRenderer
     {
 #if ASP_NET_CORE
-        private IHostEnvironment HostEnvironment => _hostEnvironment ?? (_hostEnvironment = ServiceLocator.ResolveService<IHostEnvironment>(ResolveService<IServiceProvider>(), LoggingConfiguration));
+        private IHostEnvironment RetrieveHostEnvironment()
```

Comment:
Could this method be moved down below `HostEnvironment`-property, and be changed to this:
```c#
private IHostEnvironment RetrieveHostEnvironment()
{
#if ASP_NET_CORE
    return ServiceLocator.ResolveService<IHostEnvironment>(ResolveService<IServiceProvider>(), LoggingConfiguration);
#else
    return Internal.HostEnvironment.Default
#endif
}
```